lib/module.js

```diff
@@ -489,7 +489,9 @@ Module._initPaths = function() {
 
   var nodePath = process.env['NODE_PATH'];
   if (nodePath) {
-    paths = nodePath.split(path.delimiter).concat(paths);
+    paths = nodePath.split(path.delimiter).concat(paths).filter(function(path) {
+      return !!path
+    });
```

Comment:
@silverwind @Fishrock123  `path !== ''` is slightly faster than `!!path`, but `!!path` can handle some special situation, e.g. `path == null`, in this case, `path.length !== 0` might throw exception.

```
Using node 1.7.1
!!str x 81,087,829 ops/sec ±1.46% (183 runs sampled)
str !== '' x 88,814,451 ops/sec ±0.98% (184 runs sampled)
str.length !== 0 x 87,047,827 ops/sec ±0.96% (186 runs sampled)
```